src/org/labkey/test/util/AuditLogHelper.java

```diff
@@ -69,6 +76,116 @@ public DataRegionTable goToAuditEventView(String eventType)
+                                                         List<Filter> filters, @Nullable Integer maxRows) throws IOException, CommandException
+    {
+        SelectRowsCommand cmd = new SelectRowsCommand("auditLog", auditEventName);
+        cmd.setColumns(columnNames);
+        cmd.addFilter("ProjectId/DisplayName", projectName, Filter.Operator.EQUAL);
+        filters.forEach(cmd::addFilter);
+        if (maxRows != null)
+            cmd.setMaxRows(maxRows);
+
+        String formattedContainerPath = projectName;
+        if (!formattedContainerPath.startsWith("/"))
+            formattedContainerPath = "/" + formattedContainerPath;
+        if (!folderName.equalsIgnoreCase(projectName))
+            formattedContainerPath = formattedContainerPath + "/" + folderName;
+
+        return cmd.execute(WebTestHelper.getRemoteApiConnection(), formattedContainerPath);
+    }
+
+    /**
+     * Check the number of diffs in the audit event. This is a helper function to check the number of diffs in the
+     * newRecordMap for an audit entry. If a transactionId is provided, it will check all rows for that
+     * transactionId. If no transactionId is provided, it will check just the latest row.
+     */
+    public static void checkTimelineAuditEventDiffCount(String projectName, String folderName, List<Integer> expectedDiffCounts) throws IOException, CommandException
+    {
+        checkTimelineAuditEventDiffCount(projectName, folderName, getAuditEventNameFromURL(), expectedDiffCounts);
+    }
+    public static void checkTimelineAuditEventDiffCount(String projectName, String folderName, String auditEventName, List<Integer> expectedDiffCounts) throws IOException, CommandException
+    {
+        Integer maxRows = expectedDiffCounts.size();
+        List<Map<String, Object>> events = getAuditLogsFromLKS(projectName, folderName, auditEventName, List.of("NewRecordMap"), Collections.emptyList(), maxRows).getRows();
+        for (int i = 0; i < expectedDiffCounts.size(); i++)
+        {
+            Integer expectedDiffCount = expectedDiffCounts.get(i);
+            Map<String, Object> event = events.get(i);
+            String dataChangesStr = (String) event.get("NewRecordMap");
+            String[] dataChanges = dataChangesStr != null ? dataChangesStr.split("&") : new String[0];
+            // filter out SampleStateLabel as that is not a change, it is added for display purposes
+            dataChanges = Stream.of(dataChanges).filter(s -> !s.toLowerCase().startsWith("samplestatelabel=")).toArray(String[]::new);
+            BaseWebDriverTest.getCurrentTest().log("Audit record data changes diff count check: " + dataChangesStr);
+            BaseWebDriverTest.getCurrentTest().checker().verifyEquals("Audit record data changes did not include the expected number of diffs, expected " + expectedDiffCount + " but was " + dataChanges.length + ": " + dataChangesStr,
```

Comment:
@labkey-tchad You mentioned via chat that `BaseWebDriverTest.getCurrentTest()` isn't correct here because of the test scoping. Do you have a recommended approach here? Should I just throw or should I pass through the BaseWebDriverTest in some way?

Reply:
ah, ok perfect. that makes sense. thank you

Reply:
@labkey-tchad Trey, I'm going to need some help with this. I am able to change the _wrapper in AuditLogHelper to type BaseWebDriverTest, that part makes sense. However, I am still unsure how I can then use the AuditLogHelper inside of the component class such as EntityBulkUpdateDialog or DetailTableEdit. I don't see a way that they have access to the BaseWebDriverTest. It seems like what I'd really like is for that AuditLogHelper _wrapper to be of type WebDriverWrapper, but then I wouldn't have access to the checker().

Reply:
Gotcha, that makes sense to me. I'll make it an assertEquals(). Thank you.